data_ingestion/app.py

```diff
@@ -61,16 +59,32 @@ def send_points(self, points_by_bucket):
         for bucket in points_by_bucket:
             self.write_api.write(bucket, self.org, points_by_bucket[bucket])
 
-    def to_points(self, data_dict):
-        points_by_bucket = dict()
-        for metric in data_dict:
-            metric_name = metric.split("-")
-            table, field = metric_name[0], metric_name[-1]
-            bucket = data_dict[metric]['bucket']
-            point = Point(table) \
-                .field(field, data_dict[metric]['value']) \
-                .time(data_dict[metric]['timestamp'], WritePrecision.S)
-            points_by_bucket.setdefault(bucket, []).append(point)
+    def to_points(self, message):
+
+        bucket_name = message["asset"]
+        data = message["data"]
+
+        points_by_bucket = {
+            bucket_name: []
+        }
+
+        # Unpacking data
+        for module_name, dic in data.items():
+            for timestamp, fields in dic.items():
+                # Creating a Point object with module_name as measurement
+                point = Point(module_name).time(timestamp, WritePrecision.S)
+                # Adding channel name as field key and data as value to the point
+                for channel, data in fields.items():
+                    point.field(channel, data)
+
+                # Print the InfluxDB point object as line protocol
+                line_protocol = point.to_line_protocol()
+                print("InfluxDB Point Object: ", line_protocol)
+
+                # Append it under the bucket
+                points_by_bucket[bucket_name].append(point)
+
+        print("points added to the buckets successfully.")
         return points_by_bucket
```

Comment:
We dont really have a notion of 'points_by_bucket' anymore since all points from a single message are going to the same bucket, lets refactor this so that its more clear.